httpx/_transports/urllib3.py

```diff
@@ -15,6 +15,21 @@
     urllib3 = None
 
 
+class URLLib3ByteStream(httpcore.SyncByteStream):
+    def __init__(self, conn: urllib3.HTTPResponse) -> None:
+        self._conn = conn
+
+    def __iter__(self) -> Iterator[bytes]:
+        try:
+            for chunk in self._conn.stream(4096, decode_content=False):
+                yield chunk
+        except socket.error as exc:
+            raise httpcore.NetworkError(exc) from exc
+
+    def close(self) -> None:
+        self._conn.release_conn()
```

Comment:
Changes in this module are redundant given #1182, but I've updated this all the same since this PR removes `close_func` from our internal `IteratorStream`.